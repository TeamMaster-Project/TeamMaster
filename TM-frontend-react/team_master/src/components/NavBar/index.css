.navbar {
  padding: 5px 80px;
  background-color: rgb(101, 91, 196) !important;
  background-image: linear-gradient(
    to right,
    rgb(0, 98, 255),
    rgb(7, 137, 160)
  ) !important;
  z-index: 10;
  top: 0;
  width: 100%;
  /* position: fixed !important; */
}

.nav-link {
  /* color: white !important; */
} 
