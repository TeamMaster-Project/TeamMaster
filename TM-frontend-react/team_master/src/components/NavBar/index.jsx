import React from "react";
import * as ReactBootstrap from "react-bootstrap";
import { Link, NavLink, BrowserRouter as Router } from "react-router-dom";
import "./index.css";

const NavBar = (props) => {
  //console.log(props.currentUser);
  return (
    <ReactBootstrap.Navbar
      collapseOnSelect
      expand="lg"
      bg="dark"
      variant="dark"
      className="navbar"
    >
      <ReactBootstrap.Navbar.Brand href="/">
        TeamMaster
      </ReactBootstrap.Navbar.Brand>
      <ReactBootstrap.Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <ReactBootstrap.Navbar.Collapse id="responsive-navbar-nav">
        <ReactBootstrap.Nav className="mr-auto">
          {/* <NavLink className="nav-link" to="/home">
            Home
          </NavLink> */}
           <Router>
              <NavLink className="nav-link" to="/newproject">
                New Project
              </NavLink>
           </Router>
           <Router>
              <NavLink className="nav-link" to="/myprojects">
                My Projects
              </NavLink>
            </Router>
            <Router>
              <NavLink className="nav-link" to="/chatroom">
                My Chat Rooms
              </NavLink>
            </Router>
            <Router>
              <NavLink className="nav-link" to="/videoroom">
                My Conference Rooms
              </NavLink>
            </Router>
            <Router>
              <NavLink className="nav-link" to="/contact">
                Contact Us
              </NavLink>
            </Router>
        </ReactBootstrap.Nav>

        <ReactBootstrap.Nav>
          {props.currentUser && (
            <React.Fragment>
              <li className="nav-item">
              <Router>
                <NavLink className="nav-link" to="#">
                  Hi {props.currentUser.name} !
                </NavLink>
              </Router>
              </li>
              <li className="nav-item">
              <Router>
                <NavLink className="nav-link" to="/logout">
                  Log out
                </NavLink>
              </Router>
              </li>
            </React.Fragment>
          )}
          {!props.currentUser && (
            <React.Fragment>
              <Router>
                <NavLink className="nav-link" to="/login">
                  Log In
                </NavLink>
              </Router>
              <Router>
                <NavLink className="nav-link" to="/register">
                  Register
                </NavLink>
              </Router>
            </React.Fragment>
          )}        
        </ReactBootstrap.Nav>
      </ReactBootstrap.Navbar.Collapse>
    </ReactBootstrap.Navbar>
  );
};

export default NavBar;
