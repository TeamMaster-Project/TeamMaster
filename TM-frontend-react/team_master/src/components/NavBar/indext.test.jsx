import React from 'react'
import {render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NavBar from '.';

afterEach(()=>{
    cleanup();
})

const windowOpen = jest.fn();
window.open = windowOpen;

const NavBarProps = {
    currentUser: { 
        username: "mihindu@gmail.com",
        name: "Mihindu Ranasinghe"
     }
};



test('Matches Snapshot',()=>{
    const tree = renderer.create(<NavBar/>).toJSON();
    expect(tree).toMatchSnapshot();
})

describe('Testing Navigation items based on the autherization', () => {
    it('NavBar items testing for a logged in user', async () => {
        const { getByTestId, getByText } = render(<NavBar {...NavBarProps} />);

        const NewProject = getByText("New Project");
        expect(NewProject).toBeInTheDocument();

        const MyProjects = getByText("My Projects");
        expect(MyProjects).toBeInTheDocument();

        const MyChatRooms = getByText("My Chat Rooms");
        expect(MyChatRooms).toBeInTheDocument();

        const MyConferenceRooms = getByText("My Conference Rooms");
        expect(MyConferenceRooms).toBeInTheDocument();

        const ContactUs = getByText("Contact Us");
        expect(ContactUs).toBeInTheDocument();

        const Greeting = getByText("Hi Mihindu Ranasinghe !");
        expect(Greeting).toBeInTheDocument();

        const LogOut = getByText("Log out");
        expect(LogOut).toBeInTheDocument();
    });  

    it('NavBar items testing for a logged out user', async () => {
        const { getByTestId, getByText } = render(<NavBar/>);

        const NewProject = getByText("New Project");
        expect(NewProject).toBeInTheDocument();

        const MyProjects = getByText("My Projects");
        expect(MyProjects).toBeInTheDocument();

        const MyChatRooms = getByText("My Chat Rooms");
        expect(MyChatRooms).toBeInTheDocument();

        const MyConferenceRooms = getByText("My Conference Rooms");
        expect(MyConferenceRooms).toBeInTheDocument();

        const ContactUs = getByText("Contact Us");
        expect(ContactUs).toBeInTheDocument();

        const LogIn = getByText("Log In");
        expect(LogIn).toBeInTheDocument();

        const Register = getByText("Register");
        expect(Register).toBeInTheDocument();
    });
});
