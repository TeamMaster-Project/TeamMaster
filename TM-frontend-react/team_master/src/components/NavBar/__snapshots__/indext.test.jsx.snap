// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<nav
  className="navbar navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    TeamMaster
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/newproject"
        onClick={[Function]}
      >
        New Project
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/myprojects"
        onClick={[Function]}
      >
        My Projects
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/chatroom"
        onClick={[Function]}
      >
        My Chat Rooms
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/videoroom"
        onClick={[Function]}
      >
        My Conference Rooms
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/contact"
        onClick={[Function]}
      >
        Contact Us
      </a>
    </div>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
</nav>
`;
